lib/view_component/system_test_helpers.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module ViewComponent
+  module SystemTestHelpers
+    include TestHelpers
+    include Capybara::DSL
+
+    def page
+      Capybara.current_session
+    end
+
+    def visit_rendered_component_in_browser(component, **options)
+      layout = options[:layout] || false
+
+      opts = {
+        layout: layout,
+        locals: {
+          render_args: {
+            component: component,
+            template: "view_components/preview",
+            hide_preview_source: true
+          }
+        }
+      }
+      html = controller.render_to_string("view_components/preview", opts)
+
+      # Add './tmp/view_component_integrations/' directory if it doesn't exist to store the rendered component html
+      FileUtils.mkdir_p("./tmp/view_component_integrations/") unless Dir.exist?("./tmp/view_component_integrations/")
+
+      # Write to temporary file to contain fully rendered component
+      # within a browser
+      file = Tempfile.new(["rendered_#{component.class.name}", ".html"], "tmp/view_component_integrations")
```

Comment:
If we render a component twice with different arguments, wouldn't one clobber the other here?

Reply:
@joelhawksley I believe it wouldn't because the output of `Tempfile` creates a unique string at the end. For example:
```
[1] pry(main)> Tempfile.new(["hello.html"])
=> #<File:/var/folders/ll/np7j8vfd05q58297qqhhbs1w0000gp/T/hello.html20220915-66507-tjp6pm>
[2] pry(main)> Tempfile.new(["hello.html"])
=> #<File:/var/folders/ll/np7j8vfd05q58297qqhhbs1w0000gp/T/hello.html20220915-66507-9gc2ki>
[3] pry(main)> Tempfile.new(["hello.html"])
=> #<File:/var/folders/ll/np7j8vfd05q58297qqhhbs1w0000gp/T/hello.html20220915-66507-128ga2h>
[4] pry(main)> Tempfile.new(["hello.html"])
=> #<File:/var/folders/ll/np7j8vfd05q58297qqhhbs1w0000gp/T/hello.html20220915-66507-19cjlew>
[5] pry(main)> Tempfile.new(["hello.html"])
=> #<File:/var/folders/ll/np7j8vfd05q58297qqhhbs1w0000gp/T/hello.html20220915-66507-gd95yb>
[6] pry(main)> Tempfile.new(["hello.html"])
=> #<File:/var/folders/ll/np7j8vfd05q58297qqhhbs1w0000gp/T/hello.html20220915-66507-17xubsr>
```

Reply:
Oh, that's great 👍🏻